sync/src/main/java/tech/pegasys/artemis/sync/PeerSync.java

```diff
@@ -101,10 +103,11 @@ public PeerSync(
   }
 
   private void blockResponseListener(BeaconBlock block) {
-    try {
-      blockImporter.importBlock(block);
-    } catch (StateTransitionException e) {
-      throw new BadBlockException("State transition error", e);
+    final BlockImportResult result = blockImporter.importBlock(block);
+    if (result.isSuccessful()) {
+      return;
+    } else if (result.getFailureReason() == FailureReason.FAILED_STATE_TRANSITION) {
+      throw new BadBlockException("State transition error", result.getFailureCause());
```

Comment:
This will ignore blocks that aren't connected but since we're requesting and importing blocks in order that would be a reason to consider the peer bad and disconnect them.  So I'd be inclined to throw `BadBlockException` anytime `result` is not successful.

Reply:
Updated to disconnect if the parent is unknown, and otherwise fail if the import failed for any of the remaining cases.